src/ubuntu/18.04/amd64/Dockerfile

```diff
@@ -28,9 +28,21 @@ RUN apt-get update \
         npm \
         tar \
         zip \
-    && rm -rf /var/lib/apt/lists/* \
-    && npm install -g azure-cli@0.10.15 \
-    && npm cache clean
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Azure CLI
+RUN apt-get update \
+    && apt-get install ca-certificates curl apt-transport-https lsb-release gnupg -y \
+    && curl -sL https://packages.microsoft.com/keys/microsoft.asc | \
+        gpg --dearmor | \
+        tee /etc/apt/trusted.gpg.d/microsoft.asc.gpg > /dev/null \
+    && echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" | \
+        tee /etc/apt/sources.list.d/azure-cli.list \
+    && apt-get update \
+    && apt-get install azure-cli \
+    && rm -rf /etc/apt/trusted.gpg.d/microsoft.asc.gpg \
```

Comment:
What is the reason for removing the Azure-CLI repository here?  This isn't a typical pattern I'm used to seeing.  It is beneficial to retain it (albeit most use cases are edge cases).

Reply:
I was just cleaning up the state that was modified in order to install the package.  I'm fine with keeping it there if you think that's the right way to go.

Reply:
My preference is to keep the repository registration.  It supports scenarios like apt-get upgrade.